### ISSUE TITLE
Cant declare inline annotations in xtend

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext-xtend#256]
```
class Foo  {
	
	def <T extends Number> m() {
  		val String s = xxxx(5)
	}
	
	@Inline("xxxx") def static String xxxx(int size) {
		null
	}
	
}
```

```
java.lang.IllegalStateException
	at org.eclipse.xtext.xbase.compiler.FeatureCallCompiler.appendInlineFeatureCall(FeatureCallCompiler.java:959)
	at org.eclipse.xtext.xbase.compiler.FeatureCallCompiler.appendFeatureCall(FeatureCallCompiler.java:902)
	at org.eclipse.xtext.xbase.compiler.FeatureCallCompiler.featureCalltoJavaExpression(FeatureCallCompiler.java:585)
	at org.eclipse.xtext.xbase.compiler.FeatureCallCompiler._toJavaExpression(FeatureCallCompiler.java:568)
	at org.eclipse.xtext.xbase.compiler.FeatureCallCompiler.internalToConvertedExpression(FeatureCallCompiler.java:100)
	at org.eclipse.xtext.xbase.compiler.XbaseCompiler.internalToConvertedExpression(XbaseCompiler.java:342)
	at org.eclipse.xtend.core.compiler.XtendCompiler.internalToConvertedExpression(XtendCompiler.java:425)
	at org.eclipse.xtext.xbase.compiler.TypeConvertingCompiler.internalToConvertedExpression(TypeConvertingCompiler.java:104)
	at org.eclipse.xtext.xbase.compiler.XbaseCompiler._toJavaStatement(XbaseCompiler.java:587)
	at org.eclipse.xtext.xbase.compiler.XbaseCompiler.doInternalToJavaStatement(XbaseCompiler.java:375)
	at org.eclipse.xtend.core.compiler.XtendCompiler.doInternalToJavaStatement(XtendCompiler.java:367)
	at org.eclipse.xtext.xbase.compiler.AbstractXbaseCompiler.internalToJavaStatement(AbstractXbaseCompiler.java:473)
	at org.eclipse.xtext.xbase.compiler.XbaseCompiler._toJavaStatement(XbaseCompiler.java:394)
	at org.eclipse.xtext.xbase.compiler.XbaseCompiler.doInternalToJavaStatement(XbaseCompiler.java:349)
	at org.eclipse.xtend.core.compiler.XtendCompiler.doInternalToJavaStatement(XtendCompiler.java:367)
	at org.eclipse.xtext.xbase.compiler.AbstractXbaseCompiler.internalToJavaStatement(AbstractXbaseCompiler.java:473)
	at org.eclipse.xtext.xbase.compiler.AbstractXbaseCompiler.compile(AbstractXbaseCompiler.java:294)
	at org.eclipse.xtext.xbase.compiler.AbstractXbaseCompiler.compile(AbstractXbaseCompiler.java:274)
	at org.eclipse.xtext.xbase.compiler.JvmModelGenerator.compile(JvmModelGenerator.java:1057)
	at org.eclipse.xtend.core.compiler.XtendGenerator.compile(XtendGenerator.java:213)
	at org.eclipse.xtext.xbase.compiler.JvmModelGenerator.generateExecutableBody(JvmModelGenerator.java:1034)
	at org.eclipse.xtext.xbase.compiler.JvmModelGenerator._generateMember(JvmModelGenerator.java:798)
	at org.eclipse.xtext.xbase.compiler.JvmModelGenerator.generateMember(JvmModelGenerator.java:1660)
	at org.eclipse.xtext.xbase.compiler.JvmModelGenerator.lambda$generateMembersInBody$2(JvmModelGenerator.java:292)
	at org.eclipse.xtext.xbase.lib.ObjectExtensions.operator_doubleArrow(ObjectExtensions.java:139)
	at org.eclipse.xtext.xbase.compiler.LoopExtensions.forEach(LoopExtensions.java:35)
	at org.eclipse.xtext.xbase.compiler.JvmModelGenerator.generateMembersInBody(JvmModelGenerator.java:295)
	at org.eclipse.xtend.core.compiler.XtendGenerator.generateMembersInBody(XtendGenerator.java:573)
	at org.eclipse.xtext.xbase.compiler.JvmModelGenerator._generateBody(JvmModelGenerator.java:269)
	at org.eclipse.xtext.xbase.compiler.JvmModelGenerator.generateBody(JvmModelGenerator.java:1632)
	at org.eclipse.xtext.xbase.compiler.JvmModelGenerator.generateType(JvmModelGenerator.java:226)
	at org.eclipse.xtext.xbase.compiler.JvmModelGenerator._internalDoGenerate(JvmModelGenerator.java:213)
	at org.eclipse.xtext.xbase.compiler.JvmModelGenerator.internalDoGenerate(JvmModelGenerator.java:1615)
	at org.eclipse.xtext.xbase.compiler.JvmModelGenerator.doGenerate(JvmModelGenerator.java:196)
	at org.eclipse.xtend.core.compiler.XtendGenerator.doGenerate(XtendGenerator.java:95)
	at org.eclipse.xtend.core.compiler.XtendGenerator.doGenerate(XtendGenerator.java:111)
	at org.eclipse.xtext.generator.GeneratorDelegate.doGenerate(GeneratorDelegate.java:43)
	at org.eclipse.xtext.builder.ParallelBuilderParticipant.handleChangedContents(ParallelBuilderParticipant.java:183)
	at org.eclipse.xtext.builder.ParallelBuilderParticipant.handleChangedContents(ParallelBuilderParticipant.java:172)
	at org.eclipse.xtext.builder.ParallelBuilderParticipant.doGenerate(ParallelBuilderParticipant.java:343)
	at org.eclipse.xtext.builder.ParallelBuilderParticipant$1.run(ParallelBuilderParticipant.java:308)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:111)
	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:58)
	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:75)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

```

**Comment by cdietrich** [comment]
place where the bad stack is created

```
JvmCustomAnnotationValueImpl.<init>() line: 55	
TypesFactoryImpl.createJvmCustomAnnotationValue() line: 547	
JvmTypesBuilder.toJvmAnnotationValue(XExpression) line: 1448	
JvmTypesBuilder.getJvmAnnotationReference(XAnnotation) line: 1424	
JvmTypesBuilder.addAnnotation(JvmAnnotationTarget, XAnnotation) line: 1372	
JvmTypesBuilder.addAnnotations(JvmAnnotationTarget, Iterable<XAnnotation>) line: 1358	
XtendJvmModelInferrer.translateAnnotationsTo(List<XAnnotation>, JvmAnnotationTarget) line: 365	
XtendJvmModelInferrer.transform(XtendFunction, JvmGenericType, boolean) line: 622	
XtendJvmModelInferrer.transform(XtendMember, JvmGenericType, boolean) line: 566	
XtendJvmModelInferrer.initialize(XtendClass, JvmGenericType) line: 404	
XtendJvmModelInferrer$5.run() line: 278	
XtendJvmModelInferrer$3.run() line: 218	
BatchLinkableResource.ensureJvmMembersInitialized() line: 231	
IXtendJvmAssociations$Impl(JvmModelAssociator).getOrInstall(Resource) line: 112	
IXtendJvmAssociations$Impl(JvmModelAssociator).getLogicalContainerMapping(Resource) line: 127	
IXtendJvmAssociations$Impl(JvmModelAssociator).getAssociatedExpression(JvmIdentifiableElement) line: 134	
XtendReentrantTypeResolver(LogicalContainerAwareReentrantTypeResolver).recordExpressions(JvmIdentifiableElement) line: 414	
XtendReentrantTypeResolver(LogicalContainerAwareReentrantTypeResolver).prepare(ResolvedTypes, IFeatureScopeSession) line: 408	
XtendReentrantTypeResolver(LogicalContainerAwareReentrantTypeResolver).computeTypes(ResolvedTypes, IFeatureScopeSession) line: 696	

``

**Comment by cdietrich** [comment]
see https://bugs.eclipse.org/bugs/show_bug.cgi?id=428815 as well

**Comment by cdietrich** [comment]
i will do some better handling with this issue and fix the complete handling over the bugzilla later

**Comment by cdietrich** [comment]
fixed in 2.13
